rules/prefer-query-selector.js

```diff
@@ -11,17 +11,37 @@ const disallowedIdentifierNames = new Map([
 	['getElementById', 'querySelector'],
 	['getElementsByClassName', 'querySelectorAll'],
 	['getElementsByTagName', 'querySelectorAll'],
+	['getElementsByName', 'querySelectorAll'],
 ]);
 
 const getReplacementForId = value => `#${value}`;
 const getReplacementForClass = value => value.match(/\S+/g).map(className => `.${className}`).join('');
+const getReplacementForName = (value, originQuote) => value.match(/\S+/g).map(name => `[name=${wrapQuoted(name, originQuote)}]`).join('');
 
 const getQuotedReplacement = (node, value) => {
 	const leftQuote = node.raw.charAt(0);
 	const rightQuote = node.raw.at(-1);
 	return `${leftQuote}${value}${rightQuote}`;
 };
 
+const wrapQuoted = (value, originalQuote) => {
+	switch (originalQuote) {
+		case '\'': {
+			return `"${value}"`;
+		}
+
+		case '"': {
+			return `'${value}'`;
+		}
+
+		case '`': {
+			return `'${value}'`;
+		}
+
+		// No default
+	}
+};
```

Comment:
I don't think this is safe, but I guess no one put quotes or other strange stuff in the name?

A better fix should direct update the raw string

```diff
- ('foo')
+ ('[name="foo"]')
```

```diff
- ("foo")
+ ("[name=\"foo\"]")  // Personally, I like double quoted attribute selector.
```

```diff
- (`foo`)
+ (`[name="foo"]`)
```

Anyway, if it's hard we can leave it as is for now.